mlflow/gateway/handlers.py

```diff
@@ -0,0 +1,109 @@
+from enum import Enum
+from pathlib import Path
+from pydantic import BaseModel, Field, validator, parse_obj_as
+from pydantic.json import pydantic_encoder
+from typing import Optional, Union, List
+import yaml
+import json
+
+
+class Provider(str, Enum):
+    UNSPECIFIED_PROVIDER = "unspecified"
+    OPENAI = "openai"
+    ANTHROPIC = "anthropic"
+    BARD = "bard"
```

Comment:
@dbczumar @BenWilson2 If we integrate LangChain, which provider will be used? `UNSPECIFIED_PROVIDER`?

Reply:
Another named provider. We'll want to make this plugin friendly as well